### ISSUE TITLE
FTBFS on Fedora 40

### ISSUE THREAD

**Issue by vashirov** [389ds/389-ds-base#5962]
**Issue Description**
389-ds-base fails to to build from source on Fedora 40 (Rawhide):
```
In file included from ldap/servers/slapd/back-ldbm/back-ldbm.h:75,
                 from ldap/servers/slapd/back-ldbm/dbimpl.c:33:
./ldap/servers/slapd/slap.h:544:20: error: expected ':', ',', ';', '}' or '__attribute__' before '.' token
  544 | #define f_type f_un.f_un_type
      |                    ^
./ldap/servers/slapd/slap.h:544:20: error: expected ':', ',', ';', '}' or '__attribute__' before '.' token
  544 | #define f_type f_un.f_un_type
      |                    ^
```
**Package Version and Platform:**
 - Platform: Fedora 40
 - Package and version: 2.4.x

**Comment by vashirov** [comment]
gcc versions are the same on F39 and F40:
`gcc (GCC) 13.2.1 20231011 (Red Hat 13.2.1-4)`

**Comment by vashirov** [comment]
Ok, found the cause:
> -Werror=implicit-int -Werror=implicit-function-declaration coming to rawhide

https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/HDXD72LT57KBZIS33VIBTVFKFSGG37PA/#XX6G5UTYBD2JKKEBPMRMXXFZYOHKIFNV

https://fedoraproject.org/wiki/Changes/PortingToModernC#Detailed_Description

**Comment by vashirov** [comment]
This was fixed in 9d0ebfe5c1c964526448e113dc13d10731c27992 but introduced a build failure on 32-bit arches: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1063434
